tests/kafkatest/tests/connect/connect_distributed_test.py

```diff
@@ -155,7 +155,43 @@ def test_restart_failed_connector(self):
         wait_until(lambda: self.connector_is_running(self.sink), timeout_sec=10,
                    err_msg="Failed to see connector transition to the RUNNING state")
 
-    
+    @matrix(delete_before_reconfig=[False, True])
+    def test_bad_connector_class(self, delete_before_reconfig):
+        """
+        For the same connector name, first configure it with a bad connector class name such that it fails to start, verify that it enters a FAILED state.
+        Restart should also fail.
+        Then try to rectify by reconfiguring it as a MockConnector and verifying it successfully transitions to RUNNING.
+        """
+        self.setup_services()
+        self.cc.set_configs(lambda node: self.render("connect-distributed.properties", node=node))
+        self.cc.start()
+
+        connector_name = 'bad-to-good-test'
+
+        connector = namedtuple('BadConnector', ['name', 'tasks'])(connector_name, 1)
+        config = {
+            'name': connector.name,
+            'tasks.max': connector.tasks,
+            'connector.class': 'java.util.HashMap'
```

Comment:
I have a not great reason for not using `NonExistentConnectorClass` here... if `ClassNotFoundException` is triggered in `Worker.startConnector()`, it falls back to reflection for traversing classpath to find a class matching that alias and that takes a while in system test VM's. Long enough that the 15s status check times out.

https://github.com/apache/kafka/blob/59cfa84801c67de9729385a8f9b536721e0c37b9/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java#L201